sagemaker-fundamentals/execution-role/execution-role.ipynb

```diff
@@ -0,0 +1,582 @@
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def attach_permission(role_name, policy_name, policy_doc):\n",
+    "    \"\"\"Attach a basic permission policy to the role\"\"\"\n",
+    "\n",
+    "    # Create the policy\n",
+    "    # If the policy with policy name $policy_name already exists,\n",
+    "    # then we need to delete it first\n",
+    "    \n",
+    "    # Note: you need to make sure that you do not have a policy \n",
+    "    # with $policy_name in production, because we will delete it\n",
+    "    # and create a new one with the policy document given by \n",
+    "    # $policy_doc\n",
+    "    \n",
+    "    policy = None\n",
+    "    for p in iam.list_policies()['Policies']:\n",
+    "        if p['PolicyName']==policy_name:\n",
+    "            # Before we delete the policy, we need to detach it\n",
+    "            # from all IAM entities \n",
+    "            policy = boto3.resource('iam').Policy(p['Arn'])\n",
+    "            \n",
+    "            # 1. detach from all groups\n",
+    "            for grp in policy.attached_groups.all():\n",
+    "                policy.detach_group(GroupName=grp.name)\n",
+    "                \n",
+    "            # 2. detach from all users\n",
+    "            for usr in policy.attached_users.all():\n",
+    "                policy.detach_user(UserName=usr.name)\n",
+    "            \n",
+    "            # 3. detach from all roles\n",
+    "            for rol in policy.attached_roles.all():\n",
+    "                policy.detach_role(RoleName=rol.name)\n",
+    "                \n",
+    "            break\n",
+    "    \n",
+    "    if policy is not None:\n",
+    "        iam.delete_policy(PolicyArn=policy.arn)   \n",
+    "    \n",
+    "    # create a new policy\n",
+    "    policy = iam.create_policy(\n",
+    "        PolicyName=policy_name,\n",
+    "        PolicyDocument=json.dumps(policy_doc))['Policy']\n",
+    "    \n",
+    "    # attach the policy to the role\n",
+    "    res = iam.attach_role_policy(\n",
+    "        RoleName=role_name,\n",
+    "        PolicyArn=policy['Arn']\n",
+    "        )\n",
+    "    return res"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "perm_res = attach_permission(\n",
+    "    role_name=role_res['Role']['RoleName'],\n",
+    "    policy_name='BasicS3Policy',\n",
+    "    policy_doc=basic_s3_permission\n",
+    "    )\n",
+    "\n",
+    "pp.pprint(perm_res)\n",
+    "\n",
+    "# Allow 15 seconds for the update to propagate\n",
+    "time.sleep(15)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test your role\n",
+    "\n",
+    "Now, you can verify that the role we just created (`basic-role`) is allowed to list all S3 buckets under your account and it is not allowed to do anything else with your AWS resource. "
```

Comment:
Re-word: "You can now verify..."

"....anything else with your Amazon Resource Name (ARN)."